packages/manager/apps/dedicated/client/app/dedicated/server/mrtg/mrtg.controller.js

```diff
@@ -0,0 +1,199 @@
+import get from 'lodash/get';
+import head from 'lodash/head';
+import map from 'lodash/map';
+import some from 'lodash/some';
+
+class MrtgCtrl {
+  constructor($q, $scope, $stateParams, $translate, Server) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.Server = Server;
+
+    this.server = {};
+
+    this.serverStatsLoad = {
+      loading: true,
+      error: false,
+    };
+
+    this.serverStats = {
+      model: null,
+      modelconst: null,
+      networkChoice: null,
+      periodeChoice: null,
+      typeChoice: null,
+      mrtgShow: false,
+    };
+
+    this.networks = [];
+  }
+
+  $onInit() {
+    this.$scope.$on('reloadChart', () => {
+      this.getStatistics();
+    });
+
+    this.loadStatistics();
+  }
+
+  static convertData(list) {
+    return map(list, (value) =>
+      value.unit === 'bps' ? (value.y / 1024).toFixed(2) : value.y,
+    );
+  }
+
+  createChart(data) {
+    this.series = [];
+    this.data = [];
+    this.labels = map(get(data, 'download.values'), (value) =>
+      moment.unix(value.timestamp).calendar(),
+    );
+    this.series.push(
+      this.$translate.instant('server_tab_STATS_legend_download'),
+    );
+    this.series.push(this.$translate.instant('server_tab_STATS_legend_upload'));
+    this.data.push(MrtgCtrl.convertData(get(data, 'download.values')));
+    this.data.push(MrtgCtrl.convertData(get(data, 'upload.values')));
+
+    const { unit } = head(get(data, 'download.values'));
+    const yLabel =
+      unit === 'bps'
+        ? this.$translate.instant(
+            'server_configuration_mitigation_statistics_unit_KB',
+          )
+        : unit;
+    this.options = {
+      scales: {
+        yAxes: [
+          {
+            scaleLabel: {
+              display: true,
+              labelString: yLabel,
+            },
+          },
+        ],
+        xAxes: [
+          {
+            scaleLabel: {
+              display: true,
+              labelString: this.$translate.instant(
+                'server_usage_chart_xaxis_label',
+              ),
+            },
+          },
+        ],
+      },
+    };
+  }
+
+  loadStatistics() {
+    const nameServer = null;
+    this.serverStatsLoad.loading = true;
+    this.serverStatsLoad.error = false;
+    const promises = {
+      interfaces: this.Server.getNetworkInterfaces(this.$stateParams.productId),
+      statConst: this.Server.getStatisticsConst(),
+    };
+
+    this.$q
+      .all(promises)
+      .then((stats) => {
+        const hasVrack = some(
+          stats.interfaces,
+          (networkInterface) => networkInterface.linkType === 'private',
+        );
+        if (!hasVrack) {
+          stats.interfaces.push({
+            linkType: 'no_vrack',
+            disabled: true,
+          });
```

Comment:
It's a bit weird to modify the HTTP response data. You should add it directly in `this.networks`

Reply:
done